### ISSUE TITLE
Other image formats

### ISSUE THREAD

**Issue by dnnrly** [dnnrly/layli#13]
**Is your feature request related to a problem? Please describe.**
Can we output to PNG and SVG please?

**Comment by hiitsm** [comment]
Hey. Could you assign me this issue. I'd love to try working on it

**Comment by dnnrly** [comment]
Hi, I don't assign issues to people. Feel free to discuss an approach on the issue and when you feel it is the right time, raise a PR with the change you want to make.

**Comment by hiitsm** [comment]
hey, so id wanted to take a bit to read the code and understand and i think i do now. my thouhgt process for this issue would be along the lines of using a combination of oksvg and rasterx to create the png file, partially inspired by this:
> https://stackoverflow.com/a/63227777

so i was thinking that a command be added in main for the conversion and then afterwards add a conversion function that
-takes svg input
-create and set dimensions
-converts to png
-output png file

i hope this can work and any improvements/comments would be well appreciated :3

**Comment by dnnrly** [comment]
So are you thinking that the SVG would be generated as normal then you use the additional library to convert that into PNG?

**Comment by hiitsm** [comment]
yep exactly. whilst not the best resource wise i was thinking of this approach because some people may not wanna convert so to just leave them with that option we could make it optional

**Comment by dnnrly** [comment]
OK, go for it! I'm happy to answer any questions you might have. Feel free to raise a PR early to do some Proof of Concept work, just make sure that you make it clear in the PR title.

There's no particular right answer but the kind of things that I look for are:
* Being easy to understand - try to keep complex algorithms and organisation to a minimum
* Easy to modify or extend - kind of an extension of the above point
* Try to keep external packages to a minimum - be pragmatic but understand that more packages means more maintenance burden

**Comment by hiitsm** [comment]
hey, sorry i js got the email notification for this but ill start on it right away. ill try to keep it expandable with functions and try to reduce the packages required. i'll let you know when i get a working version

**Comment by hiitsm** [comment]
hey, so i made some changes by firstly making a new file called svg_converter which does the image conversion using oksvg and rasterx then linked it in main.go and even added a command for it. luckily it does convert the SVG to PNG🥳, **but** the background and text of the converted file is not present, so i was wondering if you could help me figure out why. the files are linked properly and the process partially works but i just dont know what the issue is. i'll maybe try another means later today. sadly idk the best way to add the code here as it isn't letting me attach the files so i'll js paste it here. sorry about that and can you lmk a better way of letting you see my code?

svg_converter:

`package converter

import(
		"image"
		"image/png"
		"os"
		"fmt"

		"github.com/srwiley/oksvg"
		"github.com/srwiley/rasterx"
)

// converts SVG to image.Image
// image.Image is GO's universal img format that can be changed into wtv u want
func svgToImage(svgPath string) (image.Image, error){
	// Opens SVG file
	file, err := os.Open(svgPath)
	if err != nil {
		return nil, fmt.Errorf("Opening SVG file: %w", err)
	}
	defer file.Close()

	// Read file content
	icon, err := oksvg.ReadIconStream(file)
	if err != nil {
		return nil, fmt.Errorf("Reading SCG file: %w", err)
	}

	width := int(icon.ViewBox.W)
	height := int(icon.ViewBox.H)

	//defualt values if height and width unknown
	if width == 0 || height == 0 {
		width = 800
		height = 800
	}

	// Sets target rendering dimensions
	// Makes RGBA image "buffer"
	icon.SetTarget(0, 0, float64(width), float64(height))
	rgba := image.NewRGBA(image.Rect(0, 0, width, height))

	// Makes rasterizer and draws
	scanner := rasterx.NewScannerGV(width, height, rgba, rgba.Bounds())
	dasher := rasterx.NewDasher(width, height, scanner)
	icon.Draw(dasher, 1)

	return rgba, nil
}

// converts SVG to PNG
func SVGToPNG(svgPath, outputPath string) error {
	// convert svg to img
	img, err := svgToImage(svgPath)
	if err != nil {
		return err
	}

	// create output file
	outputFile, err := os.Create(outputPath)
	if err != nil {
		return fmt.Errorf("Creating output file: %w", err)
	}
	defer outputFile.Close()

	//cencode as PNG 
	return png.Encode(outputFile, img)
}`

main.go updated part:

rootCmd.AddCommand(
	&cobra.Command{
		Use:   "to-png [flags] [svg file]",
		Short: "convert an SVG file to a PNG file",
		Long:  `Convert an SVG file into a PNG file`,

		Args: cobra.ExactArgs(1),
		RunE: func(cmd *cobra.Command, args []string) error {
			
			// defualt option in case output is not specified by --output
			if output == "" {
				output = strings.TrimSuffix(args[0], ".svg")
				output = output + ".png"
			}

			//calls convertor to convert from SVG to PNG
			err := converter.SVGToPNG(args[0], output)
			if err != nil {
				return fmt.Errorf("converting to PNG: %w", err)
			}			

			fmt.Printf("Successfully converted " + args[0] + " to " + output)
			return nil
		},
	})

**Comment by dnnrly** [comment]
Can you create a PR with the change? It'll be easier to collaborate there.
